clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -271,6 +283,7 @@ public CompletableFuture<Void> maybeAutoCommitAllConsumedNow(
         return (response, throwable) -> {
             autoCommitState.ifPresent(autoCommitState -> autoCommitState.setInflightCommitStatus(false));
             if (throwable == null) {
+                offsetCommitCallbackInvoker.submitCommitInterceptors(allConsumedOffsets);
```

Comment:
OK, but then the name of the method should be changed to better reflect what it does.

Reply:
You asked why is "this" only called in the case of the auto commit. I think you were referring to submitCommitInterceptors. It is not only called in the case of auto-commit, but also in commitAsync.

Then you mentioned that the name of "this method" is misleading. I think you are referring to maybeAutocommitAllConsumedNow. I don't think it is, because it is only called in an auto-commit.

Reply:
OK, I see it now that `submitCommitInterceptors()` (now `enqueueInterceptorInvocation`) is also called in `commitAsync()`. I do not know why I haven't seen this before.